### PR TITLE
Light Animator system

### LINKED ISSUE #5581: Emergency lightbulbs
### Environment
- Game Version: last, build 5538
- OS Version: Windows 10

### Detailed description of bug
Emergency lightbulbs dont work and dont want to be connected to APC. Their BVold versions works greate

### LINKED ISSUE #9809: [BUG] Emergency lights are showing white light and are too bright.
### There is no other issue reporting the same problem.

- [X] I have searched the existing issues

### Game Version

_No response_

### I was

Client

### OS

Windows 8, 10, 11

### What happened?

image
No power but the lights are still white and pretty bright.

### What should have happened instead?

Red and dimmer lights.

### Steps to reproduce the issue

1. Run out of power

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MaxIsJoe**
fixes #9809
fixes #5581

This PR adds a new way to handle light animations without bloating the `lightSource` script. This also reimplements the old broken emergency lights animation that was long-lost and forgotten.

### Changelog:

CL: [New] Firelocks now have the ability to set off an emergency animation for nearby lightbulbs, this feature needs to be manually set by mappers.
CL: [New] Over-charged light sources will now flicker to indicate their burnt state. 
CL: [Improvement] Emergency lights now have their own sprites.
CL: [Improvement] Some of the rendering settings for lights has been tweaked to reduce the amount of blocky shadows a player might notice.
CL: [Fix] Emergency lights have finally been fixed after years of their mysterious disappearance (someone forgot to call the code that runs it)
CL: [Fix] Fixed a bug that would cause emergency lights to be pure white.
CL: [Fix] Fixed an edge case where light colors might not be set correctly when the client joins midway through a round.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		{
+			foreach (var anim in animations)
+			{
+				if (anim.ID == animID) PlayAnim(anim);
+				break;
+			}
+		}
+
+		[ClientRpc]
+		public void PlayAnimNetworked(int animID)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: This is one shot?

File: UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs [hunk 2]
```diff
@@ -32,25 +44,34 @@ private void OnDisable()
 
 		public void StartAnimation()
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs]: it is run on headless?

File: UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: wdym?

File: UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs]: Everything runs client side

File: UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: as in  can it be a continuous animation

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: That's up to the animation scripts themselves, but all animations currently are all continuous.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: should be a synchvar then

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: done

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: would be best to name it as play animation one shot

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: no animations are one shot

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: then remove the RPC

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs:54]: done

**Pull request merged**